desc/compute/__init__.py

```diff
@@ -52,16 +52,37 @@
 
 
 # rather than having to recursively compute the full dependencies every time we
-# compute something, its easier to just do it once for all quantities when we first
+# compute something, it's easier to just do it once for all quantities when we first
 # import the compute module.
 def _build_data_index():
     for key in data_index.keys():
-        full = {}
-        full["data"] = get_data_deps(key)
-        full["transforms"] = get_derivs(key)
-        full["params"] = get_params(key)
-        full["profiles"] = get_profiles(key)
+        full = {
+            "data": get_data_deps(key, False),
+            "transforms": get_derivs(key, False),
+            "params": get_params(key, False),
+            "profiles": get_profiles(key, False),
+        }
         data_index[key]["full_dependencies"] = full
 
+        full_with_axis_data = get_data_deps(key, True)
+        if len(full["data"]) >= len(full_with_axis_data):
```

Comment:
is this possible?

Reply:
- Short answer: only the equality is possible, I suppose the inequality was just a defensive programming habit on my part.
- Long answer: `data_index[key]["full_dependencies"]["data"]` stores all the names of the quantities that `key` depends on. `data_index[key]["full_with_axis_dependencies"]["data"]` stores all the names of the quantities that `key` and the limit at the axis of `key` depend on. The purpose of creating the extra `full_with_axis` dictionary is the same as the normal one (described in the comment at line 55).

It is possible for some quantity and all its dependencies to manifestly not depend on anything extra to evaluate its limit (e.g. `data["rho"]`). In that case, `data_index[key]["full_dependencies"]["data"]` is a list which is identical to `data_index[key]["full_with_axis_dependencies"]["data"]`. Hence, when the length of these lists are equal, we know that `data_index[key]["full_with_axis_dependencies"]["data"]` will duplicate `data_index[key]["full_dependencies"] `. So I simply assign to `data_index[key]["full_with_axis_dependencies"]["data"]` the same reference that was assigned to `data_index[key]["full_dependencies"]["data"]`